imap_processing/hi/l1a/hi_l1a.py

```diff
@@ -26,27 +27,31 @@ def hi_l1a(packet_file_path: str):
     """
     unpacked_data = decom_hi.decom_packets(packet_file_path)
 
-    # group data by appId
+    # group data by apid
     grouped_data = group_by_apid(unpacked_data)
 
     # Process science to l1a.
     processed_data = []
     for apid in grouped_data.keys():
         if apid == HIAPID.H45_SCI_CNT:
-            # TODO: Add processing for science count data
-            continue
+            logger.info(
+                "Processing histogram data for [%s] packets", HIAPID.H45_SCI_CNT.name
+            )
+            data = hist_create_dataset(grouped_data[apid])
         elif apid == HIAPID.H45_SCI_DE:
             logger.info(
-                "Processing direct event data for [%s] packets", HIAPID.H45_SCI_CNT.name
+                "Processing direct event data for [%s] packets", HIAPID.H45_SCI_DE.name
             )
 
             data = science_direct_event(grouped_data[apid])
-            processed_data.append(data)
         elif apid == HIAPID.H45_APP_NHK:
             logger.info(
                 "Processing housekeeping data for [%s] packets", HIAPID.H45_APP_NHK.name
             )
-            data = create_dataset(grouped_data[apid])
-            processed_data.append(data)
+            data = hk_create_dataset(grouped_data[apid])
+        else:
+            logger.warning(f"Encountered unexpected APID [{apid}]")
+            continue
```

Comment:
Is this OK to have happen, or should we be raising RunTimeError when this occurs to attempt to abort?

Reply:
Valid point.